samcli/lib/providers/sam_function_provider.py

```diff
@@ -78,23 +78,52 @@ def get(self, name: str) -> Optional[Function]:
         if not name:
             raise ValueError("Function name is required")
 
-        # support lookup by full_path
+        get_f = None
+
         if name in self.functions:
-            return self.functions.get(name)
+            # support lookup by full_path
+            get_f = self.functions.get(name)
+
+        if not get_f:
+            # If function is not found by full path, search through all functions
+
+            found_fs = []
+
+            for f in self.get_all():
+                if name in (f.function_id, f.name, f.functionname):
+                    found_fs.append(f)
+
+            # If multiple functions are found, only return one of them
```

Comment:
Why is returning one of the the right UX? Shouldn't we only invoke if we are sure that is the function they want to invoke?

Reply:
Here the logic is kept the same as before for backward compatibility. I added a warning there because I noticed that in the nested stack case like this:
`StackA\FunctionA`
`StackA\NestedStackB\FunctionA`
The command `sam local invoke FunctionA` will only invoke one of them, and the order is also not promised. The change here is to (1) Make sure the invoked function is the same for every run. (2) Warn the user and provide the list of matched full paths.
I think if we stop the invoking, that will possible break the behaviour of some existing invocations usages.

Reply:
Seems reasonable. Thanks for explaining this. Could be worth an in comment code for future reference.